src/live_restore/live_restore_fs.c

```diff
@@ -1395,90 +1389,240 @@ __live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_
+    size_t tmp_file_path_len = strlen(dest_path) + strlen(".lr_tmp") + 1;
+    WT_RET(__wt_calloc(session, 1, tmp_file_path_len, tmp_file_path));
+    WT_RET(__wt_snprintf(*tmp_file_path, tmp_file_path_len, "%s.lr_tmp", dest_path));
+    /* Delete any existing temporary file. Also report a warning if it existed already. */
+    bool exists = false;
+    WT_RET(os_fs->fs_exist(os_fs, (WT_SESSION *)session, *tmp_file_path, &exists));
+    if (!exists)
+        return (0);
+    __wt_verbose_info(session, WT_VERB_LIVE_RESTORE,
+      "Found existing temporary file: %s deleting it!", *tmp_file_path);
+    return (os_fs->fs_remove(os_fs, (WT_SESSION *)session, *tmp_file_path, 0));
+}
+
+/*
+ * __live_restore_fs_atomic_copy_file --
+ *     Copy a file across for live restore. We intentionally do not call the WiredTiger copy and
+ *     sync function as we are copying between layers and that function copies between two paths.
+ *     This is the same "path" from the perspective of a function higher in the stack.
+ */
+static int
+__live_restore_fs_atomic_copy_file(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs,
+  WT_FS_OPEN_FILE_TYPE type, const char *filename)
+{
+    WT_DECL_RET;
     WT_SESSION *wt_session = (WT_SESSION *)session;
+    char *buf = NULL, *source_path = NULL, *dest_path = NULL, *tmp_dest_path = NULL;
+    bool dest_closed = false;
+    size_t read_size = lr_fs->read_size;
 
-    WT_RET_NOTFOUND_OK(
-      __live_restore_fs_has_file(lr_fs, &lr_fs->destination, session, name, &dest_exist));
-    WT_RET_NOTFOUND_OK(
-      __live_restore_fs_has_file(lr_fs, &lr_fs->source, session, name, &source_exist));
-    WT_RET(__dest_has_stop_file(lr_fs, (char *)name, session, &have_stop));
+    WT_ASSERT(session, type == WT_FS_OPEN_FILE_TYPE_LOG || type == WT_FS_OPEN_FILE_TYPE_REGULAR);
+    __wt_verbose_debug2(session, WT_VERB_LIVE_RESTORE,
+      "Transferring %s file (%s) from source to dest.\n",
+      type == WT_FS_OPEN_FILE_TYPE_LOG ? "log" : "regular", filename);
 
-    if (!dest_exist && !source_exist && !LF_ISSET(WT_FS_OPEN_CREATE))
-        WT_RET_MSG(session, ENOENT, "File %s does not exist in source or destination", name);
+    wt_off_t source_size;
+    size_t len;
+    WT_FILE_HANDLE *source_fh = NULL, *dest_fh = NULL;
 
-    if (!dest_exist && have_stop && !LF_ISSET(WT_FS_OPEN_CREATE))
-        WT_RET_MSG(session, ENOENT, "File %s has been deleted in the destination", name);
+    /* Get the full source and destination file names. */
+    WT_ERR(__live_restore_fs_backing_filename(
+      &lr_fs->source, session, lr_fs->destination.home, filename, &source_path));
+    WT_ERR(__live_restore_fs_backing_filename(
+      &lr_fs->destination, session, lr_fs->destination.home, filename, &dest_path));
 
-    /* Open it in the destination layer. */
-    WT_RET(__live_restore_fs_open_in_destination(lr_fs, session, lr_fh, name, flags, !dest_exist));
+    WT_ERR(__live_restore_remove_temporary_file(
```

Comment:
Question: Is the purpose of this remove to remove any files that might have been part of the copy process before a crash?

Reply:
Yes, that is the purpose of it. We should remove any existing copy files to avoid having to think about what it means if they already exist.

Reply:
Thanks! Comment here would be great :)

Reply:
982f4170d